src/wast-parser.y

```diff
@@ -650,9 +652,10 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check LPAR BLOCK labeling_opt block RPAR catch_list {
-      $5->label = $4;
-      Expr* try_ = Expr::CreateTry($5, $7.first);
+  | try_check labeling_opt LPAR BLOCK labeling_opt block RPAR catch_list {
```

Comment:
Note: I don't know if this is what we want, in terms of labels. The try label is for the rethrow instruction, while the block label is for other control.

This may be wrong, and we want to use the same label for both cases. If you agree that a single label should be used (like the WAT case), then where should it be located?